### PR TITLE
Enable builders to pass application metadata

### PR DESCRIPTION
**Pull request by devalevenkatesh**
**Issue #:**

**Description of changes:**
- Enable builders to pass App metadata like the AppName and the AppVersion to the Chime backend so that it helps Chime in analyzing the failures for the apps according to app versions and improve the JS SDK.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
- With each run of meetingV2 demo, the appName and the appVersion will be sent to the backend.
- Tested in backend logs and Kibana if the information is populated as expected.

- Added various unit tests to cover the unexpected values test.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
- Yes.

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
- Yes, it has been reviewed and approved already.

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
- NA.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1191,6 +1192,41 @@ messagingSession.addObserver(observer);
...
 
+### Providing application metadata
+
+Amazon Chime SDK for JavaScript allows builders to provide application metadata in the meeting session configuration. This field is optional. Amazon Chime uses application metadata to analyze meeting health trends or identify common failures to improve your meeting experience. 
+
+> ⚠️ Do not pass any Personal Identifiable Information (PII).
+
+**Use case 35.** Provide application metadata to the meeting session configuration.
+
+```
```
Comment (simmkyu) [inline README.md:1203]: Add `js`.

File: README.md [hunk 2]
```diff
@@ -1191,6 +1192,41 @@ messagingSession.addObserver(observer);
...
+
+meetingSessionConfiguration.applicationMetadata = ApplicationMetadata.create({
+  appName: 'AppName',
+  appVersion: '1.0.0'
+});
+```
+
+#### Accepted application metadata constraints
+
+```
```
Comment (simmkyu) [inline README.md:1221]: nit: Adding `js` will help readability a bit as comments are grayed out.

File: README.md [hunk 3]
```diff
@@ -1191,6 +1192,41 @@ messagingSession.addObserver(observer);
...
+// The appName must be between 1-32 characters and satisfy following regular expression: /^[a-zA-Z0-9]+[a-zA-Z0-9_-]*[a-zA-Z0-9]+$/g
```
Comment (simmkyu) [inline README.md:1222]: optional: You can break down this comment into two lines, so readers with a laptop screen do not need to scroll horizontally.

Comment (devalevenkatesh) [reply README.md]: Cool, makes sense! Updated it, thanks!

File: README.md (hunk 2 above)
Comment (devalevenkatesh) [reply README.md]: Definitely, missed it. Have updated it thanks!

File: README.md (hunk 1 above)
Comment (devalevenkatesh) [reply README.md]: Added `js`. Thanks for the catch!

**Review by simmkyu — APPROVED** [review]

**Review by xuesichao — APPROVED** [review]

**Pull request merged**